state/mongodb/metadata.yaml

```diff
@@ -0,0 +1,66 @@
+# yaml-language-server: $schema=../../component-metadata-schema.json
+schemaVersion: v1
+type: state
+name: mongodb
+version: v1
+status: stable
+title: "MongoDB"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-mongodb/
+  - title: Connection options
+    url: https://www.mongodb.com/docs/manual/reference/connection-string/#std-label-connections-connection-options
+capabilities:
+  # If actorStateStore is present, the metadata key actorStateStore can be used
+  - actorStateStore
+  - crud
+  - transactional
+  - etag
+  - query
+metadata:
+  - name: server
+    # Required if host is not set
+    required: false
+    description: "The server to connect to, when using DNS SRV record. One of \"server\" and \"host\" is required."
+    example: '"server.example.com"'
+  - name: host
+    # Required if server is not set
+    required: false
+    description: "The host to connect to. One of \"server\" and \"host\" is required."
+    example: '"mongo-mongodb.default.svc.cluster.local:27017"'
```

Comment:
Is there a way to capture the "either X or Y is required" dependency in the metadata itself?

Reply:
Not yet. We specifically discussed that we wouldn’t support cross-field validations at this stage. (As that’s a slippery slope that can lead to going down rabbit holes ending in us creating a new Turing-complete programming language :) )